bpython/autocomplete.py

```diff
@@ -543,13 +546,31 @@ def get_completer(completers, cursor_offset, line, **kwargs):
         complete_magic_methods is a bool of whether we ought to complete
             double underscore methods like __len__ in method signatures
     """
-
     for completer in completers:
         matches = completer.matches(cursor_offset, line, **kwargs)
         if matches is not None:
-            return sorted(matches), (completer if matches else None)
+            if len(matches) > 0:
+                matches = sort_by_underscore(matches)
+            return matches, (completer if matches else None)
     return [], None
 
+def sort_by_underscore(matches):
+    """Sort single underscore attributes before double underscore ones.
+    """
+    matches = sorted(matches)
+    if len(matches) == 0 or len(matches) == 1:
+        return matches
+    one_underscore = None
+    for i in xrange(1, len(matches)+1):
```

Comment:
The list of matches starts with the name of the variable, etc. before the attribute (eg. ['t.__class__', 't.__format__', 't.__repr__', 't.__init__', 't._bar', 't._foo'] ) so that doesn't quite work unfortunately